pkg/config/config_new.go

```diff
@@ -0,0 +1,161 @@
+// config/config.go
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var DefaultConfigHandler = New()
+
+// ConfigHandler holds the application's configuration.
+type ConfigHandler struct {
+	atmosConfig *schema.AtmosConfiguration
+	v           *viper.Viper
+}
+
+// ConfigOptions defines options for adding a configuration parameter.
+type ConfigOptions struct {
+	FlagName     string      // Custom flag name (optional)
+	EnvVar       string      // Custom environment variable (optional)
+	Description  string      // Flag description
+	Key          string      // Key of the data in atmosConfiguration
+	DefaultValue interface{} // Default value of the data in atmosConfiguration
```

Comment:
modernize: simplify code by using modern constructs

```suggestion
	DefaultValue any // Default value of the data in atmosConfiguration
```